classes/c.oeclass

```diff
@@ -64,6 +64,12 @@ CFLAGS_OPT_SPEED	?= "-O2 \
 CFLAGS_OPT_SIZE		?= "-Os"
 CFLAGS_OPT_DEBUG	?= "-O1 -g -fno-omit-frame-pointer"
 
+# Enable -fPIC for powerpc architectures to support address
+# space larger than 24 bits
+CFLAGS_OPT_SPEED:>TARGET_CPU_powerpc += " -fPIC"
+CFLAGS_OPT_SIZE:>TARGET_CPU_powerpc += " -fPIC"
+CFLAGS_OPT_DEBUG:>TARGET_CPU_powerpc += " -fPIC"
```

Comment:
I don't think we should be adding it to these CFLAGS_OPT_* variables.

1. They are used for both build, host and target CFLAGS.  So you will be adding -fPIC to HOST_CFLAGS for both cross and sdk-cross, which I don't think was the idea.

2. Users/developers are supposed to be overriding CFLAGS_OPT_SPEED, so it probably best to not piggy-back stuff to them.

Maybe do

    HOST_CFLAGS:>HOST_CPU_powerpc += " -fPIC"
    TARGET_CFLAGS:>TARGET_CPU_powerpc += " -fPIC"

instead.  I haven't tried it, so some amount of testing is clearly a good idea ;)